### ISSUE TITLE
update PSyclone internal cut-down LFRic infrastructure to support mixed precision

### ISSUE THREAD

**Issue by rupertford** [stfc/PSyclone#1529]
The mixed precision example (lfric/eg19) won't compile as symbols such as `r_solver` and `r_solver_field_type` are not yet supported.

**Comment by hiker** [comment]
I have fixed this as part of #1978 (though #1979 still remains, i.e. the fact that `r_solver` and `r_def` atm is still both 64 bit, so it's not real mixed precision).

**Comment by hiker** [comment]
I am closing this ticket for now, since #2030 has just been merged. There is also #1980 (which might be a duplicate of this), and it does introduce the updated infrastructure files. I did reopen #1979 (which has been closed in between), since #1979 did NOT change the value of `r_solver` in `constants_mod.F90` to be 32 bit. I've added the required patch (2 lines) to #1979, and once #1980 is merged, we can quickly fix that.
